src/main/java/core/basesyntax/strategy/PurchaseHandler.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.transaction.FruitTransaction;
+
+public class PurchaseHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        int past;
+        int actualy = Storage.storage.get(transaction.getFruit());
```

Comment:
```suggestion
        int currentQuantity = Storage.storage.get(transaction.getFruit());
```

Reply:
to check if a key is present - Storage.storage.containsKey
if you `get` a non-existent key you will get null and NPE trying to compare it, not 0 
you don't need two local variables